### DISCUSSION TITLE
进行ocr的时候，det参数设置为False，效果为什么会比det=True要差

### DISCUSSION THREAD

**Discussion by shell-no-shell** [PaddlePaddle/PaddleOCR#13673] [Q&A]
进行ocr的时候，det参数设置为False，效果为什么会比det=True要差

**Comment by SWHL** [comment]
详细展开说说呗，问的有点抽象

**Reply by shell-no-shell** [reply]
paddle_ocr = PaddleOCR(use_angle_cls=True, lang="ch")
用paddle_ocr.ocr(cell_image, cls=True)对图片进行识别，能正确识别出图片里的文字
设置为det=False时，paddle_ocr.ocr(cell_image, det=False, cls=True)时，对图片进行文本识别，结果是错误的
<img width="303" alt="test_det" src="https://github.com/user-attachments/assets/b874ca62-fe6e-43c9-a929-5c7cb4573be4">
<img width="789" alt="image" src="https://github.com/user-attachments/assets/1f27a692-9ac4-4e35-bdbc-0ccd16849b6e">

**Reply by SWHL** [reply]
`det=False`是啥意思啊

**Comment by GreatV** [comment]
det=True的时候会crop图片，det=False的时候默认用的图片是crop好的，只包含一行文本。

**Reply by shell-no-shell** [reply]
我上面的例子就是只包含一行文本，我不想做detection

**Reply by GreatV** [reply]
你没发现det=True的时候，图片被crop了吗，比原图小。

**Reply by shell-no-shell** [reply]
我知道det会crop，但是我不想做这一步

**Reply by SWHL** [reply]
识别模型在训练时候，输入都是crop后的。不做det，不做crop，和识别模型训练数据不一致。也就难怪不效果差了

**Reply by shell-no-shell** [reply]
![image](https://github.com/user-attachments/assets/43fc4ecc-7817-4118-b561-c7538ee6872f)
这个图不crop，识别错误，是不是就是rec模型的效果问题

**Reply by SWHL** [reply]
不是，是使用的问题。当前rec模型输入就需要crop的

**Reply by shell-no-shell** [reply]
这么回复有点不对了吧，那为什么还需要det这个参数，如果rec就需要crop的话，那det就肯定为true；而且输入为多个图片的时候，det必须为false

**Reply by SWHL** [reply]
没看懂你啥意思，想干嘛

**Reply by Juruobudong** [reply]
> 这么回复有点不对了吧，那为什么还需要det这个参数，如果rec就需要crop的话，那det就肯定为true；而且输入为多个图片的时候，det必须为false

你想不从ocr传统流程走，想创新。。但飞桨默认就是这样的啊，检测模型标注的时候就是离文字边距很小很近的矩形框，那rec模型肯定匹配这个，除非你不用这框架
